### PR TITLE
style object syntax

### PR DESCRIPTION
**Pull request by PaulieScanlon**
Some new code snippets for how to use style object syntax and TypeScript example with interface

### CODE REVIEW

File: sections/advanced/style-objects.md [hunk 1]
```diff
@@ -24,3 +24,306 @@ render(
...
+render(
+  <div>
+    <Box/>
+    <SiblingBox/>
+    <SiblingBox/>
+  </div>
+);
+```
+
+### Pseudo classes
```
Comment (theetrain) [inline sections/advanced/style-objects.md]: `Pseudo-classes`

File: sections/advanced/style-objects.md [hunk 2]
```diff
@@ -24,3 +24,306 @@ render(
...
+
+render(
+  <div>
+    <H1>h1. heading</H1>
+    <P>p. paragraph</P>
+  </div>
+);
+```
+
+### Unit-less css values
```
Comment (theetrain) [inline sections/advanced/style-objects.md]: `Unitless CSS` values

File: sections/advanced/style-objects.md [hunk 3]
```diff
@@ -24,3 +24,306 @@ render(
...
+
+styled-components object styles work with both numbers and string values
```
Comment (theetrain) [inline sections/advanced/style-objects.md]: `Style objects` work with both `number` and string values. Just like styled tagged template literals, number values are assumed as `px` unless used where unitless values are typically accepted such as `lineHeight` and `flexGrow`.

File: sections/advanced/style-objects.md [hunk 4]
```diff
@@ -24,3 +24,306 @@ render(
...
+  <div>
+    <NumberBox />
+    <StringBox />
+  </div>
+);
+```
+
+### with TypeScript
+
+styled-components can work with TypeScript interfaces.
```
Comment (theetrain) [inline sections/advanced/style-objects.md]: `Style objects` can work with TypeScript interfaces.

File: sections/advanced/style-objects.md [hunk 5]
```diff
@@ -24,3 +24,306 @@ render(
...
+render(
+  <div>
+    <Box>
+      <DecendantBox className='decendant-box'/>
+    </Box>
+  </div>
+);
+```
+
+a styled-component can reference it's decendant by component
```
[the remark this answers is not in the document]
Comment (PaulieScanlon) [reply sections/advanced/style-objects.md]: The example here is to show how all selectors work when using class name as a selector and a component as a select. Kinda feels like both examples should stay.

**Pull request closed without merging**